src/components/ResourceDistribution.vue

```diff
@@ -0,0 +1,316 @@
+<template>
+  <div>
+    <h1 class="text-center">
+      Current leadership distribution across brokers per topic
+    </h1>
+
+    <div class="col-5 mx-auto">
+      <div class="row">
+        <div class="col-8 mx-auto my-auto">
+          <input
+            type="text"
+            class="form-control"
+            v-model="filter"
+            placeholder="filter topics.."
+            :disabled="stacked"
+          />
+        </div>
+        <div class="col-4 mx-auto">
+          <div class="form-check">
+            <input
+              v-model="stacked"
+              type="checkbox"
+              class="form-check-input"
+              id="stackedCheckBox"
+            />
+            <label class="form-check-label" for="stackedCheckBox">
+              Stacked view
+            </label>
+          </div>
+          <div class="dropdown">
+            <button
+              class="btn btn-secondary btn-sm dropdown-toggle"
+              type="button"
+              data-toggle="dropdown"
+              aria-haspopup="true"
+              aria-expanded="false"
+            >
+              display: {{ resource }}
+            </button>
+            <div class="dropdown-menu" aria-labelledby="dropdownMenuButton">
+              <a class="dropdown-item" href="#" @click="setResource('leaders')"
+                >Leaders</a
+              >
+              <a class="dropdown-item" href="#" @click="setResource('replicas')"
+                >Replicas</a
+              >
+              <a class="dropdown-item" href="#" @click="setResource('cpu')"
+                >CPU</a
+              >
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <div class="alert alert-danger" role="alert" v-if="error">
+      {{ error }}
+    </div>
+
+    <div class="row" v-bind:class="{ active: loaded }" v-show="stacked">
+      <div class="col-8 mx-auto">
+        <div class="card mx-4 mt-4">
+          <div class="card-inner">
+            <LineChart
+              :chart-data="formatStackedData"
+              :options="getStackedOptions"
+            ></LineChart>
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <div
+      class="row mx-auto"
+      v-bind:class="{ active: loaded }"
+      v-show="!stacked"
+    >
+      <div
+        class="col-4"
+        v-for="item in cachedKccData"
+        v-bind:key="item[0]"
+        v-show="item[0].toLowerCase().includes(filter.toLowerCase())"
+      >
+        <div class="card mt-4">
+          <div class="card-inner">
+            <LineChart
+              :chart-data="formatItemData(item)"
+              :options="formatItemOptions(item)"
+            ></LineChart>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import LineChart from '@/components/LineChart.vue'
+const backgroundColors = ['#ffffd9', '#edf8b1', '#c7e9b4', '#7fcdbb', '#41b6c4', '#1d91c0', '#225ea8', '#253494', '#081d58']
```

Comment:
can you help put these in store : https://github.com/linkedin/cruise-control-ui/blob/369d60d620c439307e7baaba2fecd0ab082c4e72/src/store/index.js#L40 ? so that we can re-use them.